pkg/ingester/otlp/ingest_handler_test.go

```diff
@@ -314,6 +315,181 @@ func TestSampleAttributes(t *testing.T) {
 	assert.Equal(t, "(process = firefox) ||| firefox.so 0x2e;firefox.so 0x1e 239", ss[1])
 }
 
+func TestDifferentServiceNames(t *testing.T) {
+	// Create a profile with two samples having different service.name attributes
+	// Expect them to be pushed as separate profiles
+	svc := mockotlp.NewMockPushService(t)
+	var profiles []*pushv1.PushRequest
+	svc.On("Push", mock.Anything, mock.Anything).Run(func(args mock.Arguments) {
+		c := (args.Get(1)).(*connect.Request[pushv1.PushRequest])
+		profiles = append(profiles, c.Msg)
+	}).Return(nil, nil)
+
+	otlpb := new(otlpbuilder)
+	// Create mappings for both services
+	otlpb.profile.Mapping = []*v1experimental.Mapping{{
+		MemoryStart: 0x1000,
+		MemoryLimit: 0x2000,
+		Filename:    otlpb.addstr("service-a.so"),
+	}, {
+		MemoryStart: 0x2000,
+		MemoryLimit: 0x3000,
+		Filename:    otlpb.addstr("service-b.so"),
+	}}
+
+	// Create different locations for each service
+	otlpb.profile.Location = []*v1experimental.Location{{
+		MappingIndex: 0, // service-a.so
+		Address:      0x1100,
+		Line: []*v1experimental.Line{{
+			FunctionIndex: 0,
+			Line:          10,
+		}},
+	}, {
+		MappingIndex: 0, // service-a.so
+		Address:      0x1200,
+		Line: []*v1experimental.Line{{
+			FunctionIndex: 1,
+			Line:          20,
+		}},
+	}, {
+		MappingIndex: 1, // service-b.so
+		Address:      0x2100,
+		Line: []*v1experimental.Line{{
+			FunctionIndex: 2,
+			Line:          30,
+		}},
+	}, {
+		MappingIndex: 1, // service-b.so
+		Address:      0x2200,
+		Line: []*v1experimental.Line{{
+			FunctionIndex: 3,
+			Line:          40,
+		}},
+	}}
+
+	// Add functions
+	otlpb.profile.Function = []*v1experimental.Function{{
+		Name:       otlpb.addstr("serviceA_func1"),
+		SystemName: otlpb.addstr("serviceA_func1"),
+		Filename:   otlpb.addstr("service_a.go"),
+	}, {
+		Name:       otlpb.addstr("serviceA_func2"),
+		SystemName: otlpb.addstr("serviceA_func2"),
+		Filename:   otlpb.addstr("service_a.go"),
+	}, {
+		Name:       otlpb.addstr("serviceB_func1"),
+		SystemName: otlpb.addstr("serviceB_func1"),
+		Filename:   otlpb.addstr("service_b.go"),
+	}, {
+		Name:       otlpb.addstr("serviceB_func2"),
+		SystemName: otlpb.addstr("serviceB_func2"),
+		Filename:   otlpb.addstr("service_b.go"),
+	}}
+
+	otlpb.profile.LocationIndices = []int64{0, 1, 2, 3}
+
+	// Create two samples with different service.name attributes and different stacktraces
+	otlpb.profile.Sample = []*v1experimental.Sample{{
```

Comment:
Could we add a third sample without service name? We should expect a separate profile with `service_name=unknown`